haystack/preview/components/generators/hugging_face/hugging_face_local.py

```diff
@@ -0,0 +1,136 @@
+        :param token: The token to use as HTTP bearer authorization for remote files.
+            If True, will use the token generated when running huggingface-cli login (stored in ~/.huggingface).
+            If the token is also specified in the pipeline_kwargs, this parameter will be ignored.
+        :param generation_kwargs: a dictionary containing keyword arguments to customize text generation.
+            See Hugging Face's documentation for more information.
+            https://huggingface.co/docs/transformers/main/en/generation_strategies#customize-text-generation
+            https://huggingface.co/docs/transformers/main/en/main_classes/text_generation#transformers.GenerationConfig
+        :param pipeline_kwargs: Dictionary containing keyword arguments used to initialize the pipeline.
+            These keyword arguments provide fine-grained control over the pipeline.
+            In case of duplication, these kwargs override model_name_or_path, task, device, and token init parameters.
+            See Hugging Face's documentation for more information on the available kwargs.
+            https://huggingface.co/docs/transformers/en/main_classes/pipelines#transformers.pipeline.task
+            In this dictionary, you can also include "model_kwargs" to specify the kwargs
+            for model initialization:
+            https://huggingface.co/docs/transformers/en/main_classes/model#transformers.PreTrainedModel.from_pretrained
+        """
+        transformers_import.check()
+
+        pipeline_kwargs = pipeline_kwargs or {}
+        generation_kwargs = generation_kwargs or {}
+
+        # check if the pipeline_kwargs contain the essential parameters
+        # otherwise, populate them with values from other init parameters
+        pipeline_kwargs.setdefault("model", model_name_or_path)
+        pipeline_kwargs.setdefault("token", token)
+        if device is not None and "device" not in pipeline_kwargs and "device_map" not in pipeline_kwargs:
+            pipeline_kwargs["device"] = device
+
+        # task identification and validation
+        if task is None:
+            if "task" in pipeline_kwargs:
+                task = pipeline_kwargs["task"]
+            elif isinstance(pipeline_kwargs["model"], str):
+                task = model_info(pipeline_kwargs["model"], token=pipeline_kwargs["token"]).pipeline_tag
+
+        if task not in ["text-generation", "text2text-generation"]:
+            raise ValueError(
+                f"Task '{task}' is not supported. "
+                f"The supported tasks are 'text-generation' and 'text2text-generation'."
+            )
+        pipeline_kwargs["task"] = task
+
+        # if not specified, set return_full_text to False for text-generation
+        # only generated text is returned (excluding prompt)
+        if task == "text-generation" and "return_full_text" not in generation_kwargs:
```

Comment:
Perhaps we can also set default for this one up there when we resolve task?
Something like:
```
generation_kwargs.setdefault("return_full_text", task != "text-generation")

```

Reply:
```python
        if task == "text-generation":
            generation_kwargs.setdefault("return_full_text", False)
```

this works...